libs/platform/src/lib/components/form/platform-file-uploader/platform-file-uploader.component.ts

```diff
@@ -0,0 +1,122 @@
+import { SelectionModel } from '@angular/cdk/collections';
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    EventEmitter,
+    Host,
+    Input,
+    Optional,
+    Output,
+    Self,
+    SkipSelf,
+    ViewEncapsulation
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+
+import { CollectionBaseInput } from '../collection-base.input';
+import { ContentDensity, FormFieldControl } from '../form-control';
+import { FormField } from '../form-field';
+
+export class FileUploaderInvalidChangeEvent {
+    constructor(
+        public source: PlatformFileUploaderComponent,
+        public payload: File[] // Contains selected item
+    ) {}
+}
+
+export class FileUploaderSelectionChangeEvent {
+    constructor(
+        public source: PlatformFileUploaderComponent,
+        public payload: File[] // Contains selected item
+    ) {}
+}
+
+@Component({
+    selector: 'fdp-file-uploader',
+    templateUrl: './platform-file-uploader.component.html',
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PlatformFileUploaderComponent extends CollectionBaseInput {
+    /** button value */
+    @Input()
+    buttonLabel: string;
+
+    /** value for input element. */
+    @Input()
+    buttonAriaLabel: string;
+
+    /** Whether the file input accepts drag and dropped files. */
+    @Input()
+    dragndrop = true;
+
+    /** Max file size in bytes that the input will accept. */
+    @Input()
+    maxFileSize = '';
+
+    /** Min file size in bytes that the input will accept. */
+    @Input()
+    minFileSize = '';
+
+    /** Whether the file input should accept multiple files. */
+    @Input()
+    multiple = true;
+
+    /** Accepted file extensions. Format: `'.png,.jpg'`. */
+    @Input()
+    accept: string;
+
+    /** content Density of element. cozy | compact */
+    @Input()
+    contentDensity: ContentDensity = 'cozy';
+
+    /** specifies number of files to allow to select */
+    @Input()
+    fileLimit: number;
+
+    @Output()
+    selectionChange: EventEmitter<FileUploaderSelectionChangeEvent> = new EventEmitter<
+        FileUploaderSelectionChangeEvent
+    >();
+
+    @Output()
+    invalidFileChange: EventEmitter<FileUploaderInvalidChangeEvent> = new EventEmitter<
+        FileUploaderInvalidChangeEvent
+    >();
+
+    @Input()
+    files: File[];
+
+    /**@hidden
+     * The model backing of the component. */
+    changedFiles: SelectionModel<File>;
+
+    get value(): any {
```

Comment:
we should be implementing `value`, taking it as an @Input() right? All examples currently show usage with Template Forms. I think we also need to show usage with Reactive Forms.

Reply:
unresolving this as there is still no Reactive Form examples